[W3iZhi] iP

W3iZhi opened the pull request:
# Talky Chatbot
> "Insert profound words about task management" - by super famous smart guy

This chatbot helps to keep track of your tasks
- ToDo
- Deadline
- Event

The usage is really simple:
1. Download the this [file](https://github.com/W3iZhi/ip/releases/download/A-Jar/Talky.jar).
2. Double click it.
3. The app will run and you can input your commands.  :+1:
4. Input "bye" to exit. 

**To Note**: The commands are case sensitive, fully in lowercaps

Features implemented:
-[X] Adding and Removing Tasks
-[] Reminders (Coming Soon!)

This is the main block that runs the code:
```java
  public static void main(String[] args) throws TalkyException {
      new Talky().run();
  }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

W3iZhi committed:
Changed Duke.java

W3iZhi committed:
Added echo commands

W3iZhi committed:
Added Add and List

W3iZhi committed:
Added mark and unmark

W3iZhi committed:
Added different types of task

W3iZhi committed:
semi-automated ui testing

W3iZhi committed:
Added Exception Handling

W3iZhi committed:
Added delete function

W3iZhi committed:
Add save function

W3iZhi committed:
Merge branch 'branch-Level-7'

W3iZhi committed:
Add LocalDateTime

W3iZhi committed:
Merge branch 'branch-Level-8'

W3iZhi committed:
More OOP

W3iZhi committed:
Organize into Package

W3iZhi committed:
Merge branch 'add-gradle-support'

W3iZhi committed:
Add gradle

W3iZhi committed:
Add JUnit

W3iZhi committed:
Add JavaDoc

W3iZhi committed:
Add find function

W3iZhi committed:
Coding Standard

W3iZhi committed:
Merge branch 'branch-A-CodingStandard'

W3iZhi committed:
Merge branch 'branch-Level-9'

W3iZhi committed:
Add checkstyle

W3iZhi committed:
Implement javafx GUI

winstonlimjy commented on src/main/java/Talky/Parser.java:
```diff
@@ -0,0 +1,115 @@
+     * @return command arguments
+     * @throws TalkyException
+     */
+    public static String[] commandArgs(String commandLine, String commandType) throws TalkyException {
```
Perhaps, commandArguments will be clearer?

winstonlimjy commented on src/main/java/Talky/TaskList.java:
```diff
@@ -0,0 +1,127 @@
+        return changedTask.getName();
+    }
+
+    public TaskList find(String keyword) {
```
Perhaps the method name could be more specific?

winstonlimjy commented on src/main/java/Talky/TaskList.java:
```diff
@@ -0,0 +1,127 @@
+     * @return tasklist as ui list format.
+     */
+
+    public String toListFormat() {
```
Perhaps having convert in the name will be better?

winstonlimjy reviewed (commented):
Great job! I could not find any obvious code naming violations!

kienvumrpm commented on src/main/java/Talky/Task.java:
```diff
@@ -0,0 +1,119 @@
+ * Class that acts as Task
+ */
+public class Task {
+    protected String name;
```
The variable marked, as a boolean, should be isMarked to follow the Java Coding Standard

W3iZhi committed:
Add assert

W3iZhi committed:
Improve Code Quality

W3iZhi committed:
Merge pull request #1 from W3iZhi/branch-A-Assertions

Add assert

W3iZhi committed:
Merge branch 'master' into branch-A-CodeQuality

W3iZhi committed:
Merge pull request #2 from W3iZhi/branch-A-CodeQuality

Improve Code Quality

W3iZhi committed:
Add help function to bot

Gives user way to find available commands.

Returns list of commands, what they do and their format when "help"
is entered by user.

W3iZhi committed:
Add user guide

W3iZhi committed:
Fix bugs

W3iZhi committed:
Add Image

W3iZhi committed:
Merge branch 'master' of https://github.com/W3iZhi/ip
